Detectors/MUON/MCH/Simulation/src/Response.cxx

```diff
@@ -83,17 +99,32 @@ double Response::chargefrac1d(float min, float max, double k2, double sqrtk3, do
   return 2. * k4 * (TMath::ATan(u2) - TMath::ATan(u1));
 }
 //______________________________________________________________________
-unsigned long Response::response(float charge)
+unsigned long Response::response(unsigned long adc)
 {
-  //FEE effects
-  return (unsigned long)charge;
+  //DecalibrateTrackerDigit functionality from
+  //AliMuonDigitizerV3 in aliroot
+  int fgNSigma = 5.0; //aliroot no
+  //no channel-by-channel noise map as in aliroot
+  float pedestalSigma = 0.0; //channnel noise 0.5 aliroot
+  float adc_out = adc;
+  float pedestalMean = 0;
+  float adcNoise = 0.0;
+  //TODO: parameter choices for match with aliroot
+
+  adc = TMath::Nint(adc_out + pedestalMean + adcNoise + 0.5);
+
+  if (adc_out < TMath::Nint(pedestalMean + fgNSigma * pedestalSigma + 0.5))
+    adc = 0;
+  if (adc > mMaxADC)
+    adc = mMaxADC;
+  return adc;
 }
 //______________________________________________________________________
 float Response::getAnod(float x)
 {
-  int n = Int_t(x / mInversePitch);
+  int n = Int_t(x * mInversePitch); //is it wrong?
   float wire = (x > 0) ? n + 0.5 : n - 0.5;
-  return mInversePitch * wire;
+  return wire / mInversePitch; //fixed 20.03.
```

Comment:
instead of dividing by `inverse pitch`, it would be better to multiply with  `pitch`

Reply:
Well, whether we use pitch or 1/pitch, we need both * and / by it ...

But I would remove the comments : x * mInversePitch _is_ correct  AFAIK. 
And //fixed 20.03 : what is that ?

Reply:
@aphecetche :
yes, I think also that it was correct; however: 
Float_t
AliMUONResponseV0::GetAnod(Float_t x) const
{
  /// Return wire coordinate closest to x.                                                                                     

  Int_t n = Int_t(x/Pitch());
  Float_t wire = (x>0) ? n+0.5 : n-0.5;
  return Pitch()*wire;
}
So, first I try to reproduce aliroot...

Reply:
Concerning the division: Since pitch and inverse pitch are constants, it is a one time cost to calculate and store them. This can avoid millions of divisions during runtime (depending on how often this is called of course).

Reply:
@sawenzel 
yes, I full understand this.
I am doing a commit, however, I again did some stupid mistake in the git-workflow that I need to recover that you are not swamped with unrelated commits here.

Reply:
@mwinn2 : Sorry the comment was meant as reply to @aphecetche .

Reply:
No problem.